src/dbnode/persist/fs/seek_manager.go

```diff
@@ -265,6 +290,11 @@ func (m *seekerManager) Test(id ident.ID, shard uint32, start time.Time) (bool,
 	byTime.Lock()
 	defer byTime.Unlock()
 
+	// Mark byTime seekers accessed so that open/close loop does not
+	// try to close byTime seekers if this function was called after
+	// these seekers were marked closed using CloseShardIndices()
+	byTime.state = byTimeSeekersAccessed
```

Comment:
Hmm, so currently we don't cache the seekers when there's a call to `Test`. Looks like the only place `Test` gets called is when a `DataBlockRetriever` tries to stream a block. If the test is negative, then it probably shouldn't have bothered to cache the seeker. If the test is positive though, it essentially notifies a fetch loop which ultimately calls `Borrow`, which does cache the seeker, so the end result seems to be that we only cache the seeker when we should.

Makes sense to not actually set this here?

Reply:
Test caches the seekers currently [here](https://github.com/m3db/m3/blob/1f40a47004dfaed2bf5cf2b8aab77c6a2fb6e750/src/dbnode/persist/fs/seek_manager.go#L273)
I am trying to keep that behavior consistent after the introduction of "closing a shard" concept

Reply:
Ah I see, that makes sense.